code/game/turfs/unsimulated/beach.dm

```diff
@@ -18,6 +18,6 @@
 	icon_state = "water"
 	turf_flags = TURF_FLAG_BACKGROUND | TURF_IS_WET | TURF_IS_HOLOMAP_PATH
 
-/turf/unsimulated/beach/water/Initialize(var/ml)
-	. = ..()
-	overlays += image("icon"='icons/misc/beach.dmi',"icon_state"="water2","layer"=MOB_LAYER+0.1)
+/turf/unsimulated/beach/water/New()
+	..()
+	add_overlay(image("icon"='icons/misc/beach.dmi',"icon_state"="water2","layer"=MOB_LAYER+0.1))
```

Comment:
`add_overlay()` in a turf New is unsafe, they can be instanced before subsystems are iirc.

Reply:
according to [TG init order documentation](https://github.com/tgstation/tgstation/blob/470cff519e57263dc0f7adcbbf04febd90de0088/code/game/world.dm#L25-L26) this is incorrect, `Master` is instantiated via a static var initializer, which runs before mapped atom New(). `/datum/controller/master/New()` is what instantiates the subsystems. this will always happen before mapped atoms are instantiated.